Functional.Tests/Results/ResultTests.cs

```diff
@@ -58,5 +62,93 @@ public void FailureDoesNotEqualFailureOfDifferentValue()
 
 			failureOne.Equals(failureTwo).Should().BeFalse();
 		}
+
+		[Fact]
+		public async Task TryWorksCorrectlyWithSuccess()
+		{
+			var objectObjectInput = Result.Success<object, object>(new object());
+			var objectExceptionInput = Result.Success<object, Exception>(new object());
+
+			objectObjectInput.TrySelect(o => (Object: o, Boolean: true), ex => throw new System.Exception("Should not be a failure")).Should().BeSuccessful(withFailureFactoryResult =>
```

Comment:
Are you familiar with the AssertSuccess(), AssertFailure(), and AssertValue() functions in this repo? They let you do x.SomeMethod().AssertSuccess().Should().Be("abc");

Reply:
Then you could do objectObjectInput.TrySelect(...).AssertSuccess().Should().Be...

Reply:
Looks like I deleted `AssertSuccess`, `AssertFailure` methods (only used by these tests) in favour of the FluentAssertions extensions, and these tests have just been moved from a different file (see `ResultExtensionsTests` above).  The tests I had written for the extension methods submitted for this PR were ported from a test suite using [Functional.Primitives.FluentAssertions](https://github.com/RyanMarcotte/Functional.Utilities/tree/master/src/Functional.Primitives.FluentAssertions), and I did not want to rework the tests given the sheer number of them.  I was also not sure how importing the `Functional.Primitives.FluentAssertions` package (which takes a dependency on the `Functional.Primitives` NuGet package) would affect testing in this repo, so I copied the assertion classes over instead.  Ensuring some kind of consistency within this repo seems prudent.

`AssertSuccess` and `AssertFailure` read like _actions_ (`void`) as opposed to _functions_ (`return T`), and (IMO) I found the readability awkward compared to the FluentAssertions versions.

As an aside, I can also simplify the tests while preserving intent:
```
const int VALUE = 1337;
var objectObjectInput = Result.Success<int, string>(VALUE);

// the objective of the test is to ensure the input (VALUE) is transformed somehow...
// the first line is the original code
// the second line is simpler but asserts the same thing
objectObjectInput.TrySelect(i => (Value: i, Boolean: true), ex => throw new System.Exception("Should not be a failure")).Should().BeSuccessful(x => x.Should().BeEquivalentTo((VALUE, true)));
objectObjectInput.TrySelect(i => i.ToString(), ex => throw new System.Exception("Should not be a failure")).Should().BeSuccessful(x => x.Should().Be(VALUE.ToString()));
```

Thoughts?

Reply:
I went in and made the test assertions one-liners to simplify things.  Tests all pass.